ViennaAdvantage/Areas/VA009/Models/PaymentModel.cs

```diff
@@ -4726,46 +4828,91 @@ public int GetWindowID(string WindowName, Ctx ct)
+                                        if (//_invoice.GetDateInvoiced() <= DateAcct && 
+                                            DateAcct <= _payschedule.GetDiscountDate())
+                                        {
+                                            M_Allocate.SetDiscountAmt(-1 * _discAmt);
+                                        }
+                                        else if (//_invoice.GetDateInvoiced() <= DateAcct && 
+                                            DateAcct <= _payschedule.GetDiscountDays2())
+                                        {
+                                            M_Allocate.SetDiscountAmt(-1 * _discAmt2);
+                                        }
                                     }
                                     else
                                     {
                                         M_Allocate.SetOverUnderAmt(0);
+                                        if (_dueAmt < 0)
+                                            M_Allocate.SetAmount(-1 * _dueAmt);
+                                        else
+                                            M_Allocate.SetAmount(_dueAmt);
                                     }
                                     M_Allocate.SetWriteOffAmt(0);
                                 }
                                 else
                                 {
-
+                                    if (//_invoice.GetDateInvoiced() <= DateAcct && 
+                                        DateAcct <= _payschedule.GetDiscountDate())
+                                    {
+                                            _dueAmt = _dueAmt - _discAmt;
+                                            M_Allocate.SetDiscountAmt(_discAmt);
+                                    }
+                                    else if (//_invoice.GetDateInvoiced() <= DateAcct && 
+                                        DateAcct <= _payschedule.GetDiscountDays2())
+                                    {
+                                        _dueAmt = _dueAmt - _discAmt2;
+                                        M_Allocate.SetDiscountAmt(_discAmt2);
+                                    }
                                     M_Allocate.SetAmount(_dueAmt);
-                                    M_Allocate.SetDiscountAmt(_discAmt);
                                     M_Allocate.SetOverUnderAmt(0);
                                     M_Allocate.SetWriteOffAmt(0);
 
                                     if (_doctype.GetDocBaseType() == "ARC")
                                     {
+                                        // VIS_427 BugID 5620 Set Discount Amount if applicable based on discount dates
                                         M_Allocate.SetAmount(-1 * _dueAmt);
-                                        M_Allocate.SetDiscountAmt(-1 * _discAmt);
+                                        if (//_invoice.GetDateInvoiced() <= DateAcct && 
```

Comment:
No need of if else 
we can get value from the object and set value with negative